inspire_disambiguation/app.py

```diff
@@ -0,0 +1,143 @@
+        $ python
+        >>> from inspire_disambiguation import conf
+        >>> conf['ES_HOSTNAME']
+        "localhost:9201"
+
+    ETHNICITY_DATA_PATH: Location of ethnicity data required for EthnicityEstimator
+        training.
+
+    ETHNICITY_MODEL_PATH: Location of dumped ethnicity model.
+
+    DISTANCE_MODEL_PATH: Location of dumped distance model.
+
+    CLUSTERING_N_JOBS: Number of processes to use for clustering.
+
+    DISPLAY_PROGRESS: When True displays progress bar during pairing process.
+
+    LOG_LEVEL: Minimum log level which will be displaying messages.
+
+    REDIS_PHONETIC_BLOCK_KEY: Key in redis from which script will read signature blocks
+        to re-cluster.
+
+    REDIS_URL: full redis url.
+
+    REDIS_TIMEOUT: Redis timeout.
+
+    SAMPLED_PAIRS_SIZE: The number of signature pairs we sample during training.
+
+        Since INSPIRE has ~3M curated signatures it would take too much time
+        to train on all possible pairs, so we sample ~1M pairs in such a way
+        that they are representative of the known clusters structure.
+
+        Note:
+
+            It MUST be a multiple of 12 for the reason explained in
+            :mod:`inspirehep.modules.disambiguation.core.ml.sampling`.
+
+    ES_HOSTNAME: hostname and port of elasticsearch.
+
+    ES_TIMEOUT: Timeout set for redis connection.
+
+    ES_MAX_QUERY_SIZE: Maximum size for one page of results from ES.
+        Note:
+             By default ES allows to get 10000 results at one page at once.
+    """
+
+    config_data = None
+    instance_path = os.path.dirname(os.path.abspath(__file__))
+    KEY_PREFIX = "DISAMBIGUATION_"
+
+    def __new__(cls):
+        if cls.config_data is not None:
+            return cls.config_data
+        else:
+            cls.config_data = {}
+            cls.init_config()
+            return cls.config_data
+
+    @classmethod
+    def init_config(cls):
+        cls.set_default_values()
+        cls.read_config_from_module()
+        cls.read_config_from_env()
+
+    @classmethod
+    def set_default_values(cls):
+        disambiguation_base_path = os.path.join(cls.instance_path, "disambiguation")
+        cls.config_data["BASE_PATH"] = disambiguation_base_path
+        cls.config_data["ETHNICITY_DATA_PATH"] = os.path.join(
+            disambiguation_base_path, "ethnicity.csv"
+        )
+        cls.config_data["ETHNICITY_MODEL_PATH"] = os.path.join(
+            disambiguation_base_path, "ethnicity.pkl"
+        )
+        cls.config_data["DISTANCE_MODEL_PATH"] = os.path.join(
+            disambiguation_base_path, "distance.pkl"
+        )
+        cls.config_data["CLUSTERING_N_JOBS"] = 8
+        cls.config_data["DISPLAY_PROGRESS"] = False
+        cls.config_data["LOG_LEVEL"] = "WARNING"
+        cls.config_data["REDIS_PHONETIC_BLOCK_KEY"] = "author_phonetic_blocks"
+        cls.config_data["REDIS_TIMEOUT"] = 60
+        cls.config_data["SAMPLED_PAIRS_SIZE"] = 12 * 50000
```

Comment:
```suggestion
        cls.config_data["SAMPLED_PAIRS_SIZE"] = 12 * 100000
```
seems to be the value I was using previously

Reply:
:+1: